### PR TITLE
Backend(Entity Creation, Migration, User Registration)

### PR DESCRIPTION
**Pull request by irf-does-work**

### CODE REVIEW

File: FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommand.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MediatR;
+
+namespace FeatureToggle.Application.Requests.Commands.UserCommands
+{
+    public class AddUserCommand : IRequest<AddUserResponse>
+    {
+        public string? Email { get; set; }
```
Comment (RahulRamakrishnan-git) [inline FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommand.cs:12]: Why every property is nullable?

File: FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            _userManager = userManager;
+        }
+
+        public async Task<AddUserResponse> Handle(AddUserCommand request, CancellationToken cancellationToken)
+        {
+            
+            User newUser = new User(request.Email,request.Name);
+
+            if (!request.Email.EndsWith("@geekywolf.com"))
```
Comment (RahulRamakrishnan-git) [inline FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs]: Create ```AddUserCommandValidator``` and move this logic there. Use ```FluentValidation.AbstractValidator```. Ref - https://www.c-sharpcorner.com/article/using-fluentvalidation-in-asp-net-core/

File: FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                    Success = false,
+                    Message = "Failed to create user",
+                    Errors = new List<string> { "EMAIL IS NOT A GW MAIL!" }
+                };
+            }
+
+            var result = await _userManager.CreateAsync(newUser, request.Password);
+            
+            
+            if (result.Succeeded)
```
Comment (RahulRamakrishnan-git) [inline FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs:37]: Use ternary operator and remove if/else condition

File: FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                {
+                    Success = false,
+                    Message = "Failed to create user",
+                    Errors = result.Errors.Select(e => e.Description).ToList()
+                };
+            }
+        }
+    }
+
+    public class AddUserResponse
```
Comment (RahulRamakrishnan-git) [inline FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs:60]: Move this class to a folder called DTO with file name - ```AddUserResponseDto```

File: FeatureToggle.Application/Requests/Queries/Login/LoginQueryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+using Microsoft.IdentityModel.Tokens;
+
+namespace FeatureToggle.Application.Requests.Queries.Login
+{
+    public class LoginQueryHandler : IRequestHandler<LoginQuery, int>
+    {
+        private UserManager<User> _userManager;
+        private readonly IConfiguration _configuration;
+
+        public LoginQueryHandler(UserManager<User> userManager, IConfiguration configuration)
```
Comment (RahulRamakrishnan-git) [inline FeatureToggle.Application/Requests/Queries/Login/LoginQueryHandler.cs]:
```suggestion
        public GetAuthTokenQueryHandler(UserManager<User> userManager, IConfiguration configuration)
```

File: FeatureToggle.API/Controllers/UserController.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+            _mediator = mediator;
+        }
+
+        [HttpPost]
+        public async Task<AddUserResponse> AddUser(AddUserCommand command)
+        {
+            return await _mediator.Send(command);
+        }
+
+        [HttpGet("signin")]
```
Comment (anilkumar-geeky) [inline FeatureToggle.API/Controllers/UserController.cs]: ??

Comment (anilkumar-geeky) [reply FeatureToggle.API/Controllers/UserController.cs]: Create a seperate controlller for Login

File: FeatureToggle.API/Controllers/WeatherForecastController.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
+using Microsoft.AspNetCore.Mvc;
+
+namespace FeatureToggle.API.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class WeatherForecastController : ControllerBase
```
Comment (anilkumar-geeky) [inline FeatureToggle.API/Controllers/WeatherForecastController.cs]: Remove this file

File: FeatureToggle.API/FeatureToggle.API.http [hunk 8]
```diff
@@ -0,0 +1,6 @@
+@FeatureToggle.API_HostAddress = http://localhost:5289
```
Comment (anilkumar-geeky) [inline FeatureToggle.API/FeatureToggle.API.http]: Is this file required?

File: FeatureToggle.API/WeatherForecast.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+namespace FeatureToggle.API
+{
+    public class WeatherForecast
```
Comment (anilkumar-geeky) [inline FeatureToggle.API/WeatherForecast.cs:3]: remove

File: FeatureToggle.API/appsettings.json [hunk 10]
```diff
@@ -0,0 +1,16 @@
+{
+    "Logging": {
+        "LogLevel": {
+            "Default": "Information",
+            "Microsoft.AspNetCore": "Warning"
+        }
+    },
+    "AllowedHosts": "*",
+    "ConnectionStrings": {
+        "FeatureDbContext": "Server=localhost,1430;Database=FeatureDB;User Id=sa;Password=Pass@w0rd;TrustServerCertificate=true",
```
Comment (anilkumar-geeky) [inline FeatureToggle.API/appsettings.json:10]: There will be only sinle Database

File: FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs (hunk 4 above)
Comment (anilkumar-geeky) [inline FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs:60]: Seperate class

File: FeatureToggle.Application/Requests/Queries/Login/LoginQueryHandler.cs [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        {
+            _userManager = userManager;
+            _configuration = configuration;
```
Comment (anilkumar-geeky) [inline FeatureToggle.Application/Requests/Queries/Login/LoginQueryHandler.cs]: Use OPtionsMontor

File: FeatureToggle.API/appsettings.json [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+        "UserDbContext": "Server=localhost,1430;Database=UserDB;User Id=sa;Password=Pass@w0rd;TrustServerCertificate=true;"
+    },
+    "AppSettings": {
```
Comment (anilkumar-geeky) [inline FeatureToggle.API/appsettings.json]: AppSettings -> Authentication

File: FeatureToggle.Domain/Entity/Feature Schema/Business.cs [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FeatureToggle.Domain.Entity.Custom_Schema
+{
+    public class Business
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
```
Comment (anilkumar-geeky) [inline FeatureToggle.Domain/Entity/Feature Schema/Business.cs]: private set for all domain properties

File: FeatureToggle.Domain/Entity/Feature Schema/BusinessFeatureFlag.cs [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FeatureToggle.Domain.Entity.Custom_Schema
+{
+    public class BusinessFeatureFlag
+    {
+        public int Id { get; set; }
```
Comment (anilkumar-geeky) [inline FeatureToggle.Domain/Entity/Feature Schema/BusinessFeatureFlag.cs]: same here too

File: FeatureToggle.Domain/Entity/Feature Schema/Feature.cs [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FeatureToggle.Domain.Entity.Custom_Schema
+{
+    public class Feature
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
```
Comment (anilkumar-geeky) [inline FeatureToggle.Domain/Entity/Feature Schema/Feature.cs]: here too

File: FeatureToggle.Domain/Entity/User Schema/Log.cs [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace FeatureToggle.Domain.Entity.User_Schema
+{
+    public class Log
+    {
+        public int Id { get; set; }
+        public User User { get; set; }
+        public string UserId { get; set; }
+        public BusinessFeatureFlag BusinessFeature { get; set; }
+        public int BusinessFeatureId { get; set; }
```
Comment (anilkumar-geeky) [inline FeatureToggle.Domain/Entity/User Schema/Log.cs]: private set

File: FeatureToggle.Domain/Entity/User Schema/User.cs [hunk 17]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+
+namespace FeatureToggle.Domain.Entity.User_Schema
+{
+    public class User : IdentityUser
```
Comment (anilkumar-geeky) [inline FeatureToggle.Domain/Entity/User Schema/User.cs]: What is IdentityUser

File: FeatureToggle.Domain/Entity/User Schema/User.cs [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        public string Name { get; set; }
```
Comment (anilkumar-geeky) [inline FeatureToggle.Domain/Entity/User Schema/User.cs]: private set

File: FeatureToggle.API/Controllers/UserController.cs (hunk 6 above)
Comment (irf-does-work) [reply FeatureToggle.API/Controllers/UserController.cs]: Corrected

File: FeatureToggle.API/Controllers/WeatherForecastController.cs (hunk 7 above)
Comment (irf-does-work) [reply FeatureToggle.API/Controllers/WeatherForecastController.cs]: Removed

File: FeatureToggle.Domain/Entity/User Schema/User.cs (hunk 17 above)
Comment (irf-does-work) [reply FeatureToggle.Domain/Entity/User Schema/User.cs]: Its a class of  [Microsoft.AspNetCore.Identity.EntityFrameworkCore].
that already has properties for username, password, mail, etc. 
also has built in password hashing and verifying methods

File: FeatureToggle.API/FeatureToggle.API.http (hunk 8 above)
Comment (irf-does-work) [reply FeatureToggle.API/FeatureToggle.API.http]: Removed

File: FeatureToggle.Infrastructure/Migrations/20241113075050_FeatureMig1.cs [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+    /// <inheritdoc />
+    public partial class FeatureMig1 : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.EnsureSchema(
+                name: "FeatureDB");
+
+            migrationBuilder.CreateTable(
```
Comment (RahulRamakrishnan-git) [inline FeatureToggle.Infrastructure/Migrations/20241113075050_FeatureMig1.cs:16]: Migration is not required for the tables inside FeatureContext

File: FeatureToggle.Infrastructure/Models/UserContext.cs [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+using System.Threading.Tasks;
+using FeatureToggle.Domain.Configurations;
+using FeatureToggle.Domain.Entity.User_Schema;
+using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore;
+
+
+namespace FeatureToggle.Infrastructure.Models
+{
+    public class UserContext : IdentityDbContext<User>
```
Comment (RahulRamakrishnan-git) [inline FeatureToggle.Infrastructure/Models/UserContext.cs:14]: Remove this UserContext class

File: FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommand.cs (hunk 1 above)
Comment (irf-does-work) [reply FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommand.cs:12]: have changed to 'required'

File: FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs (hunk 2 above)
Comment (irf-does-work) [reply FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs]: corrected

File: FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs (hunk 3 above)
Comment (irf-does-work) [reply FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs:37]: noted

File: FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs (hunk 4 above)
Comment (irf-does-work) [reply FeatureToggle.Application/Requests/Commands/UserCommands/AddUserCommandHandler.cs:60]: corrected

File: FeatureToggle.Infrastructure/Migrations/20241113075050_FeatureMig1.cs (hunk 19 above)
Comment (irf-does-work) [reply FeatureToggle.Infrastructure/Migrations/20241113075050_FeatureMig1.cs:16]: noted

File: FeatureToggle.Infrastructure/Models/UserContext.cs (hunk 20 above)
Comment (irf-does-work) [reply FeatureToggle.Infrastructure/Models/UserContext.cs:14]: Removed

File: FeatureToggle.Application/Requests/Queries/Login/LoginQueryHandler.cs (hunk 5 above)
Comment (irf-does-work) [reply FeatureToggle.Application/Requests/Queries/Login/LoginQueryHandler.cs]: updated
